include/libtorrent/sha1_hash.hpp

```diff
@@ -161,7 +161,7 @@ namespace aux {
 		}
 		bool operator!=(digest32 const& n) const
 		{
-			return !std::equal(n.m_number, n.m_number + number_size, m_number);
```

Comment:
I'm not necessarily sold on this transformation being an improvement. could you argue for it?